File: packages/swift/Sources/CatalystSwift/Models/CatalystCreateReportRequest.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// Request body for `POST /catalyst/v1/status/{id}/report`.
///
/// `reason` is one of "nsfw" | "tos_violation" | "harassment" | "spam" | "other".
public struct CatalystCreateReportRequest: Encodable, Sendable {
  public let reason: String
  public let description: String?

  public init(reason: String, description: String? = nil) {
    self.reason = reason
    self.description = description
  }
}

File: packages/dotnet/CatalystSharp/Models/Requests/CatalystCreateReportRequest.cs
using System.Text.Json.Serialization;

namespace CatalystSharp.Models.Requests;

// reason: "nsfw" | "tos_violation" | "harassment" | "spam" | "other"
public record CatalystCreateReportRequest(
    [property: JsonPropertyName("reason")] string Reason,
    [property: JsonPropertyName("description")] string? Description = null
);
